documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +263,302 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+**How to fix the violation**: If GUIDs are absolutely required in your service, please get sign off from the Azure API review board.
+
+**Impact on generated code**: Based on each language generator, this may affect SDK.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R1011" />R1011 HttpsSupportedScheme
+**Output Message**: Azure Resource Management only supports HTTPS scheme.
+
+**Description**: Verifies whether specification supports HTTPS scheme or not.
+
+**Why the rule is important**: All the ARM specification should support HTTPS endpoint.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2004" />R2004 NonApplicationJsonType
+**Output Message**: Please make sure that media types other than 'application/json' are supported by your service.
+
+**Description**: Verifies whether operation supports "application/json" as consumes or produces section.
+
+**Why the rule is important**: N/A.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2004" />R2004 NonApplicationJsonType
+**Output Message**: Please make sure that media types other than 'application/json' are supported by your service.
+
+**Description**: Verifies whether operation supports "application/json" as consumes or produces section.
+
+**Why the rule is important**: N/A.
+
+**How to fix the violation**: N/A.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2059" />R2059 UniqueResourcePaths
+**Output Message**: Multiple resource providers are not allowed in a single spec. More than one the resource paths were found: '{0}'.
+
+**Description**: Verifies whether more than one resource providers exists in the specification or not.
+
+**Why the rule is important**: Per the ARM guidelines, each swagger specification must contain one resource provider.
+
+**How to fix the violation**: N/A.
```

Comment:
could you write out guidance on how to fix it? should the swagger be split? should the team author with ARM team? should the names be checked?
Are there any examples that could be provided of "bad examples" so people can understand any pitfalls?